src/deprecations.jl

```diff
@@ -0,0 +1,51 @@
+using Base: depwarn
+
+function adjust_histogram(img::Union{GenericGrayImage, AbstractArray{<:Color3}},
+                           f::LinearStretching,
+                           args...; kwargs...)
+    
+    depwarn("adjust_histogram(img, LinearStretching()) is deprecated, use adjust_intensity(img, LinearStretching()) instead", :adjust_histogram)                       
+    return adjust_intensity(img, f, args...;kwargs...)
+end

```

Comment:
For these I think we can simplify it with

```suggestion
@deprecate adjust_histogram(img::AbstractArray, f::LinearStretching, args...; kwargs...) adjust_intensity(img, f, args...; kwargs...)
```

Reply:
It turns out that writing the longer variant is useful because I can then unify all the deprecation messages for the 3 types that need to change (LinearStretching, GammaCorrection and ContrastStretching) into one as follows:
```julia
function adjust_histogram!(img::Union{GenericGrayImage, AbstractArray{<:Color3}},
                           f::Union{LinearStretching, GammaCorrection, ContrastStretching},
                           args...; kwargs...)
    algo = typeof(f)
    depwarn("adjust_histogram!(img, $algo) is deprecated, use adjust_intensity!(img, $algo) instead", :adjust_histogram!)                       
    return adjust_intensity!(img, f, args...; kwargs...)                          
end
```

That should save us from a bunch of code duplication.

Reply:
It's just that in this case we don't need to manually construct the `depwarn` message info:

```julia
const DeprecatedIntensityMethods = Union{LinearStretching, GammaCorrection, ContrastStretching}
@deprecate adjust_histogram(img::AbstractArray, f::DeprecatedIntensityMethods, args...; kwargs...) adjust_intensity(img, f, args...; kwargs...)
```